scripts/pull_github_pr.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+# Script for pulling a github pull request
+# along with generating a merge commit message.
+# Example usage for pull request #6007:
+#
+# git checkout master
+# git pull
+# ./scripts/pull_github_pr.sh 6007
+
+set -e
+
+gh_hosts=~/.config/gh/hosts.yml
+jenkins_url="https://jenkins.scylladb.com"
+FORCE=$2
+ORANGE='\033[0;33m'
+NC='\033[0m'
+
+if [[ ( -z "$GITHUB_LOGIN" || -z "$GITHUB_TOKEN" ) && -f "$gh_hosts" ]]; then
+	GITHUB_LOGIN=$(awk '/user:/ { print $2 }' "$gh_hosts")
+	GITHUB_TOKEN=$(awk '/oauth_token:/ { print $2 }' "$gh_hosts")
+fi
+
+if [[ -z "$GITHUB_LOGIN" || -z "$GITHUB_TOKEN" ]]; then
+    echo 'Please set $GITHUB_LOGIN and $GITHUB_TOKEN'
+    exit 1
+fi
+
+if [ -z "$1" ]; then
+	echo Please provide a github pull request number
+	exit 1
+fi
+
+for required in jq curl; do
+	if ! type $required >& /dev/null; then
+		echo Please install $required first
+		exit 1
+	fi
+done
+
+curl() {
+    local opts=()
+    if [[ -n "$GITHUB_LOGIN" && -n "$GITHUB_TOKEN" ]]; then
```

Comment:
`GITHUB_LOGIN` and `GITHUB_TOKEN` are always set, so we don't need to check them here. or, we can do something like:
```
curl() {
    if [[ ( -z "$GITHUB_LOGIN" || -z "$GITHUB_TOKEN" ) && -f "$gh_hosts" ]]; then
        GITHUB_LOGIN=$(awk '/user:/ { print $2 }' "$gh_hosts")
        GITHUB_TOKEN=$(awk '/oauth_token:/ { print $2 }' "$gh_hosts")
     fi

     if [[ -z "$GITHUB_LOGIN" || -z "$GITHUB_TOKEN" ]]; then
        echo 'Please set $GITHUB_LOGIN and $GITHUB_TOKEN'
        exit 1
    fi

    command curl --user "${GITHUB_LOGIN}:${GITHUB_TOKEN}" "$@"
}
```

as these two variables are "closer" to this function, and we only set them once they are set by this function.

Reply:
I agree that there's no need to check these variables again (we checked above that they must be set, if they aren't, the script exited). I also agree that it's nicer to just use crystal-clear shell syntax like you suggested instead of the bizarre arrays used by Avi's code.
But I wouldn't move the check for these variables into the function - I think it's better to check them up-front in the beginning of the script, like Avi did.